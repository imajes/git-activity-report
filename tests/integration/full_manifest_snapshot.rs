use test_support;

#[test]
fn full_manifest_snapshot() {
  test_support::init_tracing();
  test_support::init_insta();
  let _env = test_support::with_env(&[("TZ", "UTC")]);
  let repo = test_support::fixture_repo();
  let repo_path = repo.to_str().unwrap();
  let outdir = tempfile::TempDir::new().unwrap();
  let out_path = outdir.path().to_str().unwrap();

  let mut cmd = test_support::cmd_bin("git-activity-report");
  let out = cmd
    .args([
      "--split-apart",
      "--for",
      "every month for the last 2 months",
      "--repo",
      repo_path,
      "--out",
      out_path,
      "--include-merges",
      "--include-unmerged",
      "--tz",
      "utc",
      "--now-override",
      "2025-09-01T12:00:00",
    ])
    .output()
    .unwrap();
  assert!(out.status.success());

  let top: serde_json::Value = serde_json::from_slice(&out.stdout).unwrap();
  let dir = top.get("dir").unwrap().as_str().unwrap();
  let pointer = top
    .get("manifest")
    .and_then(|v| v.as_str())
    .or_else(|| top.get("file").and_then(|v| v.as_str()))
    .expect("pointer file or manifest");
  let path = std::path::Path::new(dir).join(pointer);
  let data = std::fs::read(&path).unwrap();
  let mut v: serde_json::Value = serde_json::from_slice(&data).unwrap();
  if let Some(obj) = v.as_object_mut() {
    obj.insert("repo".into(), serde_json::Value::String("<repo>".into()));
    obj.insert("generated_at".into(), serde_json::Value::String("[generated]".into()));
  }
  insta::assert_json_snapshot!(v, { ".authors" => insta::sorted_redaction() });
}
