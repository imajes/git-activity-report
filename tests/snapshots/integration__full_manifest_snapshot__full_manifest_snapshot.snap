---
source: tests/integration/full_manifest_snapshot.rs
assertion_line: 46
---
{
  "generated_at": "[generated]",
  "include_merges": true,
  "include_patch": false,
  "include_unmerged": true,
  "ranges": [
    {
      "file": "report-2025-07.json",
      "label": "2025-07",
      "range": {
        "since": "2025-07-01T00:00:00",
        "until": "2025-08-01T00:00:00"
      }
    },
    {
      "file": "report-2025-08.json",
      "label": "2025-08",
      "range": {
        "since": "2025-08-01T00:00:00",
        "until": "2025-09-01T00:00:00"
      }
    }
  ],
  "repo": "<repo>",
  "split_apart": true
}
